pkg/api/addons.go

```diff
@@ -752,6 +758,13 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		// Remove deprecated "kube-proxy-daemonset addon"
 		o.KubernetesConfig.Addons = append(o.KubernetesConfig.Addons[:i], o.KubernetesConfig.Addons[i+1:]...)
 	}
+
+	// Enable pod-security-policy addon during upgrade to 1.15 or greater scenarios, unless explicitly disabled
+	if isUpgrade && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.15.0") && !o.KubernetesConfig.IsAddonDisabled(common.PodSecurityPolicyAddonName) {
```

Comment:
Being a bit conservative here to allow folks who definitely don't want the aks-engine PodSecurityPolicy implementation to explicitly disable it, and for upgrade to honor that.